table/block_based/block_based_table_builder.cc

```diff
@@ -1588,6 +1594,7 @@ void BlockBasedTableBuilder::WriteFilterBlock(
 void BlockBasedTableBuilder::WriteIndexBlock(
     MetaIndexBuilder* meta_index_builder, BlockHandle* index_block_handle) {
   IndexBuilder::IndexBlocks index_blocks;
+  CachableEntry<Block>* cached_entry = nullptr;
```

Comment:
Also holding on to block data for slightly longer than necessary could plausibly affect performance.